### PR TITLE
Allow localhost XDS address

### PR DESCRIPTION
**Pull request by howardjohn**
This aligns with the non-proxy XDS and CSR codepaths

### CODE REVIEW

**Review by ramaraochavali — APPROVED** [review]

File: pkg/istio-agent/xds_proxy.go [hunk 1]
```diff
@@ -504,6 +505,11 @@ func (p *XdsProxy) getTLSDialOption(agent *Agent) (grpc.DialOption, error) {
 	// strip the port from the address
 	parts := strings.Split(agent.proxyConfig.DiscoveryAddress, ":")
 	config.ServerName = parts[0]
+	// For debugging on localhost (with port forward)
+	// This matches the logic for the CA; this code should eventually be shared
+	if strings.Contains(config.ServerName, "localhost") {
+		config.ServerName = "istiod.istio-system.svc"
```
Comment (hzxuzhonghu) [inline pkg/istio-agent/xds_proxy.go:511]: I am not sure why the server name is  "istiod.istio-system.svc" not "localhost" when it is on localhost.

Comment (hzxuzhonghu) [reply pkg/istio-agent/xds_proxy.go:511]: IMO we should cleanup all these kinds of hack

Comment (howardjohn) [reply pkg/istio-agent/xds_proxy.go:511]: @hzxuzhonghu if you are running over localhost (local developement, or proxy running in same pod as istiod) the cert is signed for `istiod`, but we request localhost. This is a hack, sure, but we cannot remove it without a viable way to run over localhost

**Pull request merged**